aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,795 @@
+            retrieve_shape: List[int] = []
+            use_selected_or_np_map: Dict[str, Optional[int]] = {}
+            for dim, index_op in zip(retrieve_dims, retrieve_indices):
+                if (
+                    dim
+                    not in [
+                        DimensionNames.SpatialY,
+                        DimensionNames.SpatialX,
+                        CZI_SAMPLES_DIM_CHAR,
+                    ]
+                    and dim in dims_shape.keys()
+                ):
+                    # Handle slices
+                    if index_op is None:
+                        # Store the dim for later to inform to use the np index
+                        use_selected_or_np_map[dim] = None
+                        if dim == DimensionNames.MosaicTile:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.MosaicTile][1]
+                            )
+                        elif dim == DimensionNames.Time:
+                            retrieve_shape.append(dims_shape[DimensionNames.Time][1])
+                        elif dim == DimensionNames.Channel:
+                            retrieve_shape.append(dims_shape[DimensionNames.Channel][1])
+                        elif dim == DimensionNames.SpatialZ:
+                            retrieve_shape.append(
+                                dims_shape[DimensionNames.SpatialZ][1]
+                            )
+
+                    # Handle non-chunk dimensions (specific indices / ints)
+                    else:
+                        # Store the dim for later to inform to use the provided index
+                        use_selected_or_np_map[dim] = index_op
+                        retrieve_shape.append(1)
+
+            # Create list of planes that we will add each plane to, later we reshape
+            # Create empty arr with the desired shape to enumerate over the np index
+            planes: List[np.ndarray] = []
+            np_array_for_indices = np.empty(tuple(retrieve_shape), dtype=object)
+            for np_index, _ in np.ndenumerate(np_array_for_indices):
+                # Get each plane's index selection operations
+                # If the dimension is None, use the enumerated np index
+                # If the dimension is not None, use the provided value
+                plane_indices: Dict[str, int] = {"S": scene}
+
+                for dim in [
+                    DimensionNames.MosaicTile,
+                    DimensionNames.Time,
+                    DimensionNames.Channel,
+                    DimensionNames.SpatialZ,
+                ]:
+                    indexes = CziReader._dim_helper(
+                        dim,
+                        use_selected_or_np_map,
+                        np_index,
+                        retrieve_dims,
+                    )
+                    if indexes is not None:
+                        plane_indices[dim] = indexes
```

Comment:
A bit confused... Shouldn't this be _any_ other non plane dim? What happens if you have a CZI with a `P` or `R` dim. This won't pick that up I don't believe. In which case I think this should just be: "for dim in open czi dims, if dim not in plane dims (YXA)"